sdk/python/feast/utils.py

```diff
@@ -256,3 +256,34 @@ def _convert_arrow_to_proto(
         created_timestamps = [None] * table.num_rows
 
     return list(zip(entity_keys, features, event_timestamps, created_timestamps))
+
+
+def has_all_tags(
+    object_tags: dict[str, str], requested_tags: Optional[dict[str, str]] = None
+) -> bool:
+    if requested_tags is None:
+        return True
+    return all(object_tags.get(key, None) == val for key, val in requested_tags.items())
+
+
+def tags_list_to_dict(
+    tags_list: Optional[list[str]] = None,
+) -> Optional[dict[str, str]]:
+    if not tags_list:
+        return None
+    tags_dict: dict[str, str] = {}
+    for tags_str in tags_list:
+        tags_dict.update(tags_str_to_dict(tags_str))
+    return tags_dict
+
+
+def tags_str_to_dict(tags: str = "") -> dict[str, str]:
+    tags_list = (
+        str(tags).strip().strip("()").replace('"', "").replace("'", "").split(",")
```

Comment:
is `str(tags)` needed if `tags` is already an `str`?

Reply:
not anymore! :) . this was oddly necessary when the input was -
```python
def tags_str_to_dict(tags: Optional[str] = None)
```
given the recent changes i've made around supporting the multiple `--tags` cli option, this can go away. thanks